packages/commons/src/logging/loggerMiddleware.ts

```diff
@@ -31,46 +33,72 @@ const getLoggerMetadata = (): SessionMetaData => {
       };
 };
 
-export const customFormat = winston.format.printf(
-  ({ level, message, timestamp }) => {
-    const { userId, organizationId, correlationId } = getLoggerMetadata();
+const logFormat = (
+  msg: string,
+  timestamp: string,
+  level: string,
+  userId: string | undefined,
+  organizationId: string | undefined,
+  correlationId: string | undefined,
+  serviceName: string = ""
+) =>
+  `${timestamp} ${level.toUpperCase()} [${serviceName}] - [UID=${userId}] [OID=${organizationId}] [CID=${correlationId}] ${msg}`;
 
+export const customFormat = (serviceName?: string) =>
+  winston.format.printf(({ level, message, timestamp }) => {
+    const { userId, organizationId, correlationId } = getLoggerMetadata();
     const lines = message
       .toString()
       .split("\n")
-      .map(
-        (line: string) =>
-          `${timestamp} ${level.toUpperCase()} - [UID=${userId}] [OID=${organizationId}] [CID=${correlationId}] ${line}`
+      .map((line: string) =>
+        logFormat(
+          line,
+          timestamp,
+          level,
+          userId,
+          organizationId,
+          correlationId,
+          serviceName
+        )
       );
     return lines.join("\n");
-  }
-);
+  });
 
-export const logger = winston.createLogger({
-  level: config.logLevel,
-  transports: [
-    new winston.transports.Console({
-      stderrLevels: ["error"],
-    }),
-  ],
-  format: winston.format.combine(
-    winston.format.timestamp(),
-    winston.format.json(),
-    winston.format.errors({ stack: true }),
-    customFormat
-  ),
-  silent: process.env.NODE_ENV === "test",
-});
+const getLogger = (serviceName?: string) =>
+  winston.createLogger({
+    level: config.logLevel,
+    transports: [
+      new winston.transports.Console({
+        stderrLevels: ["error"],
+      }),
+    ],
+    format: winston.format.combine(
+      winston.format.timestamp(),
+      winston.format.json(),
+      winston.format.errors({ stack: true }),
+      customFormat(serviceName)
+    ),
+    silent: process.env.NODE_ENV === "test",
+  });
 
-export const loggerMiddleware = expressWinston.logger({
-  winstonInstance: logger,
-  requestWhitelist:
-    config.logLevel === "debug" ? ["body", "headers", "query"] : [],
-  ignoredRoutes: ["/status"],
-  responseWhitelist:
-    config.logLevel === "debug" ? ["body", "statusCode", "statusMessage"] : [],
-});
+export const loggerMiddleware = (serviceName: string) => () =>
+  expressWinston.logger({
+    winstonInstance: getLogger(serviceName),
+    requestWhitelist:
+      config.logLevel === "info" ? ["body", "headers", "query"] : [],
+    ignoredRoutes: ["/status"],
+    responseWhitelist:
+      config.logLevel === "info" ? ["body", "statusCode", "statusMessage"] : [],
+    meta: false,
+    msg: (req, res) => {
+      if (res.statusCode >= 400) {
+        // sdd
+      }
```

Comment:
?? I don't get the meaning of this if

Reply:
it's a leftover, this check could be skipped because the format remains equals for request with error and successfully; the error body was logged by [during problem creation](https://github.com/pagopa/interop-be-monorepo/blob/be748b961b0ba0644e5134bbdf672fbaddddff30/packages/models/src/errors.ts#L78) so in the message definition we can skip overhead with additional check and logs the request in the same way.

❓  A part of that, I don't know if could be useful to log the HTTP  response with status code  >=400 with error leve instead info. Anyway this improvement it's not related to `msg` property definition but with different.